### ISSUE TITLE
custom CA certificates are not persistent

### ISSUE THREAD

**Issue by ktdreyer** [rpm-software-management/mock#588]
### Short description of the problem

Mock installs the host's custom CA certificates in such a way that they are almost always wiped out later. This means that DNF/Yum cannot verify HTTPS URLs signed by a custom CA, etc.

### Output of `rpm -q mock`

mock-2.2-1.el8.noarch

### Steps to reproduce issue

1. Start with a new CentOS 8 system and install mock
2. Add a new PEM-formatted CA file to `/etc/pki/ca-trust/source/anchors/my-custom-ca.crt`
3. Trust this CA system wide: `update-ca-trust extract`
4. Confirm that this new CA is available in the system bundle:
   `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`
5. Initialize a chroot, like epel-8-x86_64:

```
mock -r epel-8-x86_64 init
```
### Actual results:

My custom CA is not present in the main epel-8-x86_64 or epel-8-x86_64-bootstrap chroot locations.

```
grep -l My-Custom-CA /var/lib/mock/epel-8-x86_64*/root/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt
```

### Expected results:
My custom CA is present in both the epel-8-x86_64 and epel-8-x86_64-bootstrap chroot locations.

### Additional information:

Stepping through `package_manager.py`, Mock is calling `copy_certs()`, and we are succesfully calling `copy_tree()` like so:
```
copy_tree("/etc/pki/ca-trust/extracted", "/var/lib/mock/epel-8-x86_64/root/etc/pki/ca-trust/extracted")
```
The problem seems to be that something later overwrites `/var/lib/mock/epel-8-x86_64/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` in the chroot.

Perhaps one of the packages in CentOS 8 overwrites `ca-bundle.trust.crt` after mock modifies it. For example, this could happen if any of the packages in the chroot execute the `update-ca-trust extract` command.

If I run `mock -r epel-8-x86_64 init` a second time without scrubbing the cache, I see mock running `copy_tree()` again, and this time the custom CA certificates from the host remain in the epel-8 and epel-8 bootstrap chroots. When I remove the cache with `mock -r epel-8-x86_64 --scrub=all`, and re-run `mock init`, I don't see the custom CA anymore. This makes me speculate that there is some RPM scriptlet that is overwriting our extracted `ca-bundle.trust.crt` file.

I tested changing `copy_certs()` to copy the entire parent directory `/etc/pki/ca-trust` into the chroot. With this change, the chroot environment has both the `extracted` and `source` directories, and if/when any scriptlet calls `update-ca-trust extract`, we'll still trust the custom CA. This solution works for both EL7 and EL8-based chroots.

**Comment by ktdreyer** [comment]
I looked through the packages in the chroot, and I found that the main "ca-certificates" package executes `/usr/bin/update-ca-trust` during `%post`. It's been this way since RHEL 6.5 when this was introduced in http://bugzilla.redhat.com/544376.

It's also customary for third-party packages to drop their own CA certificates into this location and then run `/usr/bin/update-ca-trust`. For example Red Hat has its own internal `redhat-ca-certificate` package that does this for the internal RH IT root CA.

**Comment by ktdreyer** [comment]
Note, this was released in Mock 2.4, available in EPEL 7 and 8 for several months now.

**Comment by derekschrock** [comment]
Should this also include `/usr/share/pki/` as welll?  Red Hat's [docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-shared-system-certificates_security-hardening) and [update-ca-trust](https://manned.org/update-ca-trust/e1b1e94d) use both.

During bootstrap the CA will be present however when ca-certificates is installed (and updated?) the %post will clear out since it's not present.
